resources/js/contact.js

```diff
@@ -0,0 +1,292 @@
+// https://youtu.be/eg4e-FObyJ8
+const submitBtn = document.querySelector(".formBtn")
+const fields = document.querySelectorAll(".input");
+const form = document.querySelector('form')
+
+var isValid = true
+
+submitBtn.addEventListener('click', function(){
+  validateFields(fields)
+  
+  if (isValid) {
+  
+    var contactData = {
+      name: fields[0].value,
+      email: fields[3].value,
+      phone: fields[2].value,
+      message: fields[4].value,
+    };
+    
+    const contactJSON = JSON.stringify(contactData);
+    // clearError()
+    apiCall(contactJSON);
+  } else {
+    isValid = true;
+  }
+  
+})
+
+function validateFields(fields){
+  
+  for(i=0; i<fields.length;i++){
+    var input = fields[i]
+    validate(input)
+  }
+}
+
+function validate(input) {
+  var validName = false;
+  var validEmail = false;
+  var validPhone = false;
+  var validMessage = false;
+  if (input.required && input.value.length == 0) {
+    setError(input, `${input.name} field cannot be empty!`);
+  } else {
+    if (input.id == "name") {
+      validName = validateName(input)
+    }
+    else if(input.id == "phone"){
+      validPhone = validatePhone(input)
+    }
+    else if(input.id == "email"){
+      validEmail = validateEmail(input)
+    }
+    else if(input.id == "message"){
+      validMessage = validateMessage(input)
+    }
+  }
```

Comment:
@Thejus5 ~~Missing max-chara validation for Name and Message/Description~~

Found maxlength validation in HTML and min-length in JS. Use single type of validations, either all HTML or all JS for maintainability and singular UX